src/query/functions-v2/src/scalars/variant.rs

```
@@ -43,4 +75,756 @@ pub fn register(registry: &mut FunctionRegistry) {
+        if s.is_empty() {
+            bitmap.push(false);
+            builder.commit_row();
+            continue;
+        }
+        let index = unsafe { index_arg.index_unchecked(idx) };
+        match get_by_index(s, index as usize) {
+            Some(val) => {
+                builder.put_slice(val.as_slice());
+                bitmap.push(true);
+            }
+            None => bitmap.push(false),
+        }
+        builder.commit_row();
+    }
+    match len {
+        Some(_) => {
+            let col = Column::Nullable(Box::new(NullableColumn {
+                validity: bitmap.into(),
+                column: Column::Variant(builder.build()),
+            }));
+            Ok(Value::Column(col))
+        }
+        _ => match bitmap.pop() {
+            Some(is_not_null) => {
+                if is_not_null {
+                    Ok(Value::Scalar(Scalar::Variant(builder.build_scalar())))
+                } else {
+                    Ok(Value::Scalar(Scalar::Null))
+                }
+            }
+            None => Ok(Value::Scalar(Scalar::Null)),
+        },
+    }
+}
+
+fn get_by_name_fn(
+    args: &[ValueRef<AnyType>],
+    _: FunctionContext,
+) -> Result<Value<AnyType>, String> {
+    let len = args.iter().find_map(|arg| match arg {
+        ValueRef::Column(col) => Some(col.len()),
+        _ => None,
+    });
+
+    let source_arg = args[0].try_downcast::<VariantType>().unwrap();
+    let name_arg = args[1].try_downcast::<StringType>().unwrap();
+    let size = len.unwrap_or(1);
+    let mut bitmap = MutableBitmap::with_capacity(size);
+    let mut builder = StringColumnBuilder::with_capacity(size, 0);
+    for idx in 0..size {
+        let s = unsafe { source_arg.index_unchecked(idx) };
+        if s.is_empty() {
+            bitmap.push(false);
+            builder.commit_row();
+            continue;
+        }
+        let name = unsafe { name_arg.index_unchecked(idx) };
+        let name = String::from_utf8_lossy(name);
```

Should we throw an error on this case?